recipes/autoconf/all/conanfile.py

```diff
@@ -38,7 +38,9 @@ def requirements(self):
         self.requires("m4/1.4.19") # Needed at runtime by downstream clients as well
 
     def package_id(self):
-        self.info.clear()
+        del self.info.settings.arch
```

Comment:
The ``self.info.clear()`` also clears:
- Possible variability from ``options``, not an issue in this case
- Possible variability from ``dependencies``. In this case, it could be a problem as ``self.requires("m4/1.4.19")`` is a requirement, and changing that version will create another ``package_id``. Most like the ``self.info.requires.clear()`` is necessary too

Reply:
I understand and I added this change to the branch.
I would imagine that a new version of m4 can change the output, but then the package revision would change as well.
A change in the version of m4 would also mean a change in the recipe revision though, so a new package would be generated anyway, does this actually avoid duplication?